src/lava/lib/optimization/problems/coefficients.py

```diff
@@ -53,3 +53,16 @@ def get_coefficient(self, order: int):
     def max_degree(self):
         """Maximum order among the coefficients' ranks."""
         return max(self.coefficients.keys())
+
+    def __call__(self, x: np.ndarray) -> np.ndarray:
+        result = 0
```

Comment:
First of all, I would appreciate if you could add a docstring.

But more importantly, I was wondering about the reason to use a __call__ here. Intuitively, I would expect this code in a @property method. Could you briefly elaborate why you decided for a __call__ here? You may be right that this is the best way, I just want to ensure that I understand.

Reply:
Thanks, I added the docstring.

The __call__ choice seems natural to me to express the concept of evaluating a polynomial at a given point, and given the general principle of using Pythonic methods when possible, I chose not to add an additional one for this purpose. Please let me know if you have in mind different designs, for now let's resolve this comment.

Reply:
Hm...
Then my other point would be: Wouldn't it make more sense to add the evaluation of a cost to the class Cost, instead of the CoefficientTensorsMixin?

Reply:
Thanks for your elaboration.
The CoefficientTensorsMixin is defined as "Coefficients for a scalar function of a vector.". If I call this class, I would not expect to get the evaluation of the polynomial at a specific point as return. I would rather expect to get the coefficient tensor, if anything at all. I could, however, imagine that you return the evaluation in the following forms:
- as method, maybe CoefficientTEnsorsMixin.evaluate(x)
- as __call__ for Cost

Does that make sense?